tests/Silverback.Integration.Kafka.TestConsumer/Program.cs

```diff
@@ -1,109 +1,129 @@
-        {
-            var loggerFactory = new LoggerFactory()
-                .AddNLog(new NLogProviderOptions { CaptureMessageTemplates = true, CaptureMessageProperties = true });
-            NLog.LogManager.LoadConfiguration("nlog.config");
-
-            return loggerFactory;
-        }
-    }
-}
+﻿// Copyright (c) 2018-2019 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using NLog.Extensions.Logging;
+using Silverback.Integration.Kafka.Messages;
+using Silverback.Messaging;
+using Silverback.Messaging.Broker;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Serialization;
+using Silverback.Messaging.Diagnostics;
+
+namespace Silverback.Integration.Kafka.TestConsumer
+{
+    internal static class Program
+    {
+        private static IBroker _broker;
+        private static IConsumer _consumer;
+
+        private static void Main()
+        {
+            Console.Clear();
+
+            PrintHeader();
+            Setup();
+
+            Connect();
+            Console.CancelKeyPress += (_, e) => { Disconnect(); };
+
+            while (true)
+                Console.ReadLine();
+        }
+
+        private static void Connect()
+        {
+            SilverbackDiagnostics.Enable();
+
+            var messageKeyProvider = new MessageKeyProvider(new[] { new DefaultPropertiesMessageKeyProvider() });
+            _broker = new KafkaBroker(messageKeyProvider, GetLoggerFactory(), new MessageLogger(messageKeyProvider));
+
+            var serializer = new JsonMessageSerializer<TestMessage>();
+            serializer.Settings.TypeNameHandling = TypeNameHandling.None;
+
+            _consumer = _broker.GetConsumer(new KafkaConsumerEndpoint("Topic1")
+            {
+                Configuration = new KafkaConsumerConfig
+                {
+                    BootstrapServers = "PLAINTEXT://kafka:9092",
+                    GroupId = "silverback-consumer",
+                    AutoOffsetReset = Confluent.Kafka.AutoOffsetReset.Earliest
+                },
+                Serializer = serializer
+            });
+
+            _consumer.Received += OnMessageReceived;
+
+            _broker.Connect();
+        }
+
+        private static void Setup()
+        {
+
+        }
+
+        private static void Disconnect()
+        {
+            _broker.Disconnect();
+        }
+
+        private static void OnMessageReceived(object sender, MessageReceivedEventArgs args)
+        {
+            var testMessage = args.Message as TestMessage;
+            var testMessageHeaders = args.Headers;
+
+            if (testMessage == null)
+            {
+                Console.WriteLine("Received a weird message!");
+                return;
+            }
+
+            // TODO: Add correlationId to NLog-output instead of message.

```

Comment:
You know how to do this? I only know how to do this with Serilog..

Reply:
BTW...just found the "reply" button... 🤦‍♂